ls1intum/Artemis
# Plagiarism checks for programming exercises

Use JPlag and return the html output as zip file.
Fixes #1781 

### Checklist

### Motivation and Context
We would like to simplify plagiarism checks for programming exercises

### Description
This can be done using JPlag with one click now.

Note: I left some TODOs for future improvements. We would also like to show the results directly on Artemis in the future, but this will be much more additional effort.

### Steps for Testing
1. Create and participate in programming exercises (Java, C, Python) with multiple users
2. Navigate into the `View` screen for the programming exercise
3. Scroll down and click on `Check plagiarism`
4. This will take a few seconds (or minutes for many participants)
5. Unzip the downloaded zip file and check the index.html for valid entries.

Please also double check that exporting submissions for text, modeling, file upload and programming exercises still works. I refactored some code and it should not break the functionality!

### Screenshots
![image](https://user-images.githubusercontent.com/744067/89391899-1417c100-d709-11ea-8f46-7c75e518c6f6.png)
